### DISCUSSION TITLE
Agreements use-cases

### DISCUSSION THREAD

**Discussion by xPi2** [nation3/court#12] [Ideas]
Nation3 Court Agreements WIP specs: https://github.com/nation3/court/discussions/10

To be able to refine a more solid spec for Nation3 agreement contracts I would like to collect use-cases for this kind of contracts from the community. So far we have the following:

### Role compliance agreements

Ideally required for jurors, rewards signers, community stewards and other DAO responsibility roles.
You sign some terms / code of conduct and deposit a required collateral.
Arbitrated by the DAO.
If you breach the agreed terms the DAO can liquidate your collateral.

### Sponsorships

A mechanism proposed to allow new contributors to access to the token gated areas of the organization through a loan of escrowed NATION (veNATION).
(Details to be discussed)

### Commissions / assignments

Useful to run paid tasks / project's budgets.
One party deposits a payment for a task, the other party deposits colateral.
If the task is completed under what was agreed the contract is settled the recipient can collect the payment and the colateral.
If not, any of the parties can raise a dispute and the arbitrator (for example, the Court) will resolve.
If the arbitrator resolves in favor of the payer party they can collect both the payment and the colateral from the other party.

---

**Share here other ideas or comments about these ones.**

**Comment by gregfromstl** [comment]
Bob has hired Alice to do some development work. Alice has never worked with Bob and would like some assurance that she will get paid, but Bob has the same concerns about Alice completing the work and does not want to pay upfront.

Both Bob and Alice are willing to stake* 1x the agreed payment if they can have security the other party will act in accordance with the agreement.

Creating:
One party should create the agreement with the amount they're willing to stake*
Another party (that is not the creator) should "accept" the agreement

Resolving:
One party should resolve the agreement
Another party (not the resolving party) should "accept" the resolving of the agreement

Disputes:
One party should dispute the agreement
An arbitrator should freely access the staked* amount + terms of the agreement
An arbitrator should "rule" on the agreement in favor of one party
The party ruled in favor of should "collect" their damages
Note: "Stake" in this context only means this amount will be at risk if the staking party fails to fulfill the terms of the agreement. It assumes nothing about the funds being "locked" during the agreement.
